.github/workflows/render-draft-release-notes.yml

```diff
@@ -0,0 +1,48 @@
+name: Render Draft Release Notes
+on:
+  push:
+    branches: [ dev, 'v[0-9]+.[0-9]+.x' ]
+jobs:
+  render:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Curl rpchangelog
+        run: |
+          mkdir -v -p rpchangelog
+          curl -s -S -f -L -o rpchangelog/requirements.txt https://vectorized-public.s3.us-west-2.amazonaws.com/rpchangelog/requirements.txt
+          curl -s -S -f -L -o rpchangelog/rpchangelog.py https://vectorized-public.s3.us-west-2.amazonaws.com/rpchangelog/rpchangelog.py
+          chmod +x rpchangelog/rpchangelog.py
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip'
+      - run: pip3 install -r ./rpchangelog/requirements.txt
+      - name: Render draft release notes to job summary
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          if [[ $GITHUB_REF_NAME == 'dev' ]]; then
+            PREVIOUS_TAG_NAME=$(gh -R "$GITHUB_REPOSITORY_OWNER/redpanda" release list --exclude-drafts -L 1 | tail -1 | cut -f1)
+            YEAR_VER=$(echo "$PREVIOUS_TAG_NAME" | cut -d. -f1)
+            FEATURE_VER=$(echo "$PREVIOUS_TAG_NAME" | cut -d. -f2)
+            TAG_NAME="$YEAR_VER.$((FEATURE_VER+1)).1"
+          else
+            # e.g. GITHUB_REF_NAME=v22.3.x
+            SEARCH_PATTERN=${GITHUB_REF_NAME::-1}
+            PREVIOUS_TAG_NAME=$(gh -R "$GITHUB_REPOSITORY_OWNER/redpanda" release list --exclude-drafts | grep "$SEARCH_PATTERN" | head -1 | cut -f1)
```

Comment:
discussed this during a video call. for completeness adding the tldr here: it might not always be true that this fetches the latest release, depending on whether the `latest` pointer is properly set by whoever is clicking the `publish` button on the github release UI. so it might be better to sort this and determine this via code or leverage an existing action such as this https://github.com/actions-ecosystem/action-get-latest-tag

Reply:
had a look at `action-get-latest-tag` and its computation may be thrown off by the placeholder `v23.1.0-dev` tag we use on the redpanda repo. i've created a separate tracking issue to refactor this bit of code: https://github.com/redpanda-data/vtools/issues/1183